explainer.md

```diff
@@ -215,11 +224,189 @@ defaults.
+be rewritten as an allow-lists of equal or shorter length by removing any
+block-list element from the allow-list, or dropping it altogether if there
+was no matching allow-list entry in the first place.
+
+```js
+// equivalent to:  { allowElements: ["i"] }
+const config_that_mixes_allow_and_block_lists = {
+    allowElements: ["i", "u"],
+    blockElements: ["u", "s"],
+};
+
+// <div><i>A</i></div>
+element.setHTML("<i>A</i><u>B</u><s>C</s><b>D</b>",
+                { sanitizer: config_that_mixes_allow_and_block_lists });
+```
+
+A more useful combination of different list types would be a config that
+allows simple text structure and styling, but retains the full text:
+
+```js
+const config_that_retains_simple_styling_but_most_text = {
+  allowElements: ["p", "b", "i"],
+  dropElements: ["div", "span", "em", "u", "s", "li"],
+};
+const styled_text = "<p>Some <span style='color: blue'>colourful</span> <u>styled</u> <b>text</b>";
+
+// <div><p>Some colourful styled <b>text</b></p></div>
+element.setHTML(styled_text, {sanitizer: config_that_retains_simple_styling_but_most_text});
+```
+
+### Configuring attributes per element
+
+A common use case is to allow or block all instances of a given attribute,
+but this isn't always sufficient. One may also act on attributes on specific
+elements.
+
+In the example `config_text_with_only_simple_styling` in the previous chapter
+we have allowed the `src` attribute on any of the eight allowed elements.
+If one wanted to allow `class` everywhere, but `src` only on `<img>`, the
+following would do:
+
+```js
+const config_with_element_specific_attributes = {
+  allowElements: [
+    "p", "b", "i",
+    { name: "img", attributes: [ "src" ] }
+  ],
+  allowAttributes: ["class"],
+};
+```
+
+If you want to drop `src` attributes from `<input>` elements but retain them
+elsewhere, you can use:
+
+```js
+const drop_src_attribute_from_input = {
+  dropElements: [{ name: "input", attributes: ["src"]}],
+}
+```
+
+TODO: What does `{ name: "bla", attributes: [] }` mean? My gut feeling would
+  be that `attributes: []` is a no-op and it allows `<bla>` plus whatever
+  attributes are allowed globally. But if so, the only way to describe an
+  element without any attributes in presence of a global `allowAttributes`
+  would be to add a dropElements for it.
+
+TODO: I find the syntax weird. In the allowElements case, the `img` entry
+  matches (and hence allows) both the element and the attribute. In the drop
+  case it matches only the attribute (but not the element). If we were to
+  define this identically in both cases - always match element and attributes -
+  then it simply wouldn't work for drop, because a drop-rule with attributes
+  would automatically drop the element (and hence all attributes with it).
+
+How do multiple attribute rules interact? An attribute is allowed, if at
+least one allow-rule (global or per element) allows it, and no drop
+rule drops it.
```

Comment:
I think we would go through the lists in order. If it is dropped, then we don't continue looking into the allow list. Similar to existing spec text. No?

Reply:
Discussion from today's meeting to explain the issue at hand a bit better.

Example: allowing `{name: "div", attribute: "foo"}, {name: "div", attribute: "bar"}`and input `<div foo="" bar="">`

If the algorithm would go through each entry individually, leading to `<div>`.
If the algorithm would collect all entries and combine it would return the input.